### ISSUE TITLE
Prerender broken on 1.6.2 using ExecJS?

### ISSUE THREAD

**Issue by mdolon** [reactjs/react-rails#509]
I recently upgraded react-rails from version 1.0 to the latest build 1.6.2 and a few things broke. I've been able to fix most of it (namely the server-rendering bug when assets are precompiled using [this fix](https://github.com/reactjs/react-rails/issues/443#issuecomment-180544359)).

I'm running into another problem now though when I try to use ExecJS as my renderer:
**Pass `code:` option to instantiate a JS context!**

Here's the stack trace:

```
Started GET "/" for 127.0.0.1 at 2016-04-02 13:33:06 -0400
  ActiveRecord::SchemaMigration Load (0.8ms)  SELECT "schema_migrations".* FROM "schema_migrations"
Processing by HomeController#index as HTML
  Rendered shared/_current_user.json.jbuilder (17.5ms)
  Rendered home/index.json.jbuilder (27.2ms)
  Rendered home/index.haml within layouts/application (37.4ms)
Completed 500 Internal Server Error in 54ms (ActiveRecord: 0.0ms)

ActionView::Template::Error (Pass `code:` option to instantiate a JS context!):
  app/views/home/index.haml:1:in `_app_views_home_index_haml__995446140396903069_70103800158160'

  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/actionpack-4.2.3/lib/action_dispatch/middleware/templates/rescues/_source.erb (13.1ms)
  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/actionpack-4.2.3/lib/action_dispatch/middleware/templates/rescues/_trace.html.erb (4.7ms)
  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/actionpack-4.2.3/lib/action_dispatch/middleware/templates/rescues/_request_and_response.html.erb (1.4ms)
  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/actionpack-4.2.3/lib/action_dispatch/middleware/templates/rescues/template_error.html.erb within rescues/layout (39.5ms)
  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/web-console-2.3.0/lib/web_console/templates/_markup.html.erb (0.9ms)
  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/web-console-2.3.0/lib/web_console/templates/_inner_console_markup.html.erb within layouts/inlined_string (0.8ms)
  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/web-console-2.3.0/lib/web_console/templates/_prompt_box_markup.html.erb within layouts/inlined_string (0.9ms)
  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/web-console-2.3.0/lib/web_console/templates/style.css.erb within layouts/inlined_string (0.7ms)
  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/web-console-2.3.0/lib/web_console/templates/console.js.erb within layouts/javascript (38.8ms)
  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/web-console-2.3.0/lib/web_console/templates/main.js.erb within layouts/javascript (0.7ms)
  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/web-console-2.3.0/lib/web_console/templates/error_page.js.erb within layouts/javascript (0.8ms)
  Rendered /Users/md/.rvm/gems/ruby-2.0.0-p247/gems/web-console-2.3.0/lib/web_console/templates/index.html.erb (69.5ms)
```

Here's the setup I'm using in **application.rb**:

```
    config.react.addons = true
    config.react.server_renderer = React::ServerRendering::ExecJSRenderer
    config.react.camelize_props = true
    config.react.server_renderer_options = {
      files: ["server_rendering.js"], # files to load for prerendering
    }
```

And here's the code in my view (home/index.haml):
`= react_component('HomeIndex', render(template: "home/index.json"), { prerender: true })`

Anyone run into this issue since upgrading react-rails? This code used to work before and still does work using Sprockets, but I'd rather use the speedier ExecJS renderer if possible. Any help is appreciated, many thanks!

**Comment by rmosolgo** [comment]
`SprocketsRenderer` expects a `files:` option, do you mean to use `SprocketsRenderer`? (That's the default.) 

If you want to use `ExecJSRenderer`, you should pass a `code:` option instead, containing the JavaScript code to load before serving pre-rendered components. 

In this way `ExecJSRenderer` is more "low-level": it doesn't have any methods of finding or creating JavaScript code. If you want to use the asset pipeline, you can use `SprocketsRenderer`. Otherwise, you have to fetch that code yourself somehow and pass it to `ExecJSRenderer`!

**Comment by mdolon** [comment]
Thank you for your response, @rmosolgo. I've tried it with the `code` option but haven't been able to get it to work. I've tried accessing the file using both `Rails.application.assets[file].to_s` and `ActionView::Base.new.asset_path(file)` with no luck, as sprockets-rails seems to have removed that functionality.

Since ExecJS is one of the supported renderers mentioned in the docs, is there any example code somewhere that highlights how to properly set it up?
